src/plugins/form/form.plugin.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from "@angular/core";
+import { NgxsPlugin } from "../../symbols";
+import { getTypeFromInstance, setValue } from "../../internals";
+
+@Injectable()
+export class NgxsFormPlugin implements NgxsPlugin {
+  constructor() {}
+
+  handle(state, event, next) {
+    const type = getTypeFromInstance(event);
+
+    let nextState = state;
+
+    if (type === "UpdateFormValue" || type === "UpdateForm") {
+      nextState = setValue(nextState, `${event.payload.path}.model`, {
+        ...event.payload.value
+      });
+    }
+
+    if (type === "UpdateFormStatus" || type === "UpdateForm") {
+      nextState = setValue(
+        nextState,
+        `${event.payload.path}.status`,
+        event.payload.status
+      );
+    }
+
+    if (type === "UpdateFormErrors" || type === "UpdateForm") {
+      nextState = setValue(nextState, `${event.payload.path}.errors`, {
+        ...event.payload.errors
+      });
+    }
+
+    if (type === "UpdateFormDirty" || type === "UpdateForm") {
+      nextState = setValue(
+        nextState,
+        `${event.payload.path}.dirty`,
```

Comment:
Not sure how this is working.

Reply:
this was the reducer code in ngrx-form. It looks fine to me but let me know if it can be improved?